vnext/Microsoft.ReactNative/JsiApi.idl

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.ReactNative
+{
+  interface IJsiRuntime;
+
+  delegate void JsiDataHandler(UInt8[] data);
+
+  [webhosthidden]
+  interface IJsiBuffer
+  {
+    UInt32 Size { get; };
+    void GetData(JsiDataHandler handler);
+  };
+
+  enum JsiValueKind
+  {
+    Undefined,
+    Null,
+    Boolean,
+    Number,
+    Symbol,
+    String,
+    Object,
+  };
+
+  [webhosthidden]
+  struct JsiValueData
+  {
+    JsiValueKind Kind;
+    UInt64 Data;
+  };
+
+  struct JsiSymbolData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiStringData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiObjectData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiWeakObjectData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiPropertyNameIdData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiFunctionData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiArrayData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiArrayBufferData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiScopeState
+  {
+    UInt64 Data;
+  };
+
+  [default_interface]
+  runtimeclass JsiPreparedJavaScript
```

Comment:
>JsiPreparedJavaScript [](start = 15, length = 21)

it this supposed to be empty?

Reply:
Yes, this is by design. We just need an opaque value that represents a prepared JS script. The only part is really used here is the ref-counting.

---
In reply to: [455237431](https://github.com/microsoft/react-native-windows/pull/5461#discussion_r455237431) [](ancestors = 455237431)